src/DotLiquid/Util/NumericConverter.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace DotLiquid.Util
+{
+    /// <summary>
+    /// ExtensionMethods for converting to numeric values.
+    /// </summary>
+    public static class NumericConverter
+    {
+        private static bool IsReal(object o) => o is double || o is float || o is decimal;
+        private static bool IsInteger(object o) => o is int || o is uint || o is long || o is ulong || o is short || o is ushort || o is byte || o is sbyte;
+        private static bool IsNumeric(object o) => IsReal(o) || IsInteger(o);
+
+        /// <summary>
+        /// Coerce an object into a decimal or double value
+        /// </summary>
+        /// <param name="value">The value to coerce. This can be a numeric type, or a string representation of a numeric type.</param>
+        /// <param name="formatProvider">The format provider for converting floating point numbers.</param>
+        /// <param name="defaultValue">The value to return if coercion fails.</param>
+        /// <returns>The coerced value as decimal or double type, or <paramref name="defaultValue"/> if coercion fails.</returns>
+        public static object CoerceToReal(this object value, IFormatProvider formatProvider, decimal defaultValue)
+        {
+            object result = defaultValue;

```

Comment:
consider just returning the default value directly in the fall back cases so that the code is immutable.

Reply:
Deleted.